### DISCUSSION TITLE
How to use two accounts simultaneously

### DISCUSSION THREAD

**Discussion by tcboeira** [abraunegg/onedrive#2031] [Q&A] [answered]
Hello dears,
I hope all is well with everyone.

Well, my question may seem simple but I really can't configure it so that it works successfully.

I have two accounts, one personal (outlook.com) and one company (company.com) and I want the OneDrive of both to work together like in Windows.

I followed the tips in the link (https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md), but I don't know where the error is.
- I created the directories...
```
mkdir /home/USER/.Onedrive_Config/personal
mkdir /home/USER/.Onedrive_Config/company
```
- I created files of each configuration
```
sudo nano /home/USER/.Onedrive_Config/personal/config
sudo nano /home/USER/.Onedrive_Config/company/config
```
- Where the content of "/home/USER/.Onedrive_Config/personal/config" is
```
sync_dir = "/home/user/Onedrive-Personal"
```
and the contents of "/home/USER/.Onedrive_Config/empresa/config" is
```
sync_dir = "/home/user/Onedrive-COMPANY"
skip_file = "~*|.~*|*.tmp|*.iso"
```

Now the problems begin...
When I ask to see the settings, as below, from the staff, the content is right for me:
```
onedrive --user-agent PERSONAL@outlook.com --syncdir /home/USER/Onedrive-Personal/ --confdir="/home/user/.Onedrive_Config/personal/config" --display-config
sync_dir = "/home/user/Onedrive-Personal"
```

But when I want to see what it would be for the company, the line that talks about skip_file doesn't show the *.iso information for example.

So... I really don't know what I'm doing wrong. And I don't even know how to make sure that each account downloads its content in the informed directory and that I keep it synchronized without having to change the settings.

It is possible?

In advance, thank you very much for your attention and everyone who somehow contributed to these doubts.

Hugs

**Comment by abraunegg** [comment, marked as answer]
@tcboeira 
> But when I want to see what it would be for the company, the line that talks about skip_file doesn't show the *.iso information for example.

You have listed all your personal `--display-config` output ... but nothing detailing what your company output actually is ... 

So a couple of items:
1. `--confdir` needs the configuration path ... not the file ... you are adding the full path to the file and including the file ...
2. `--syncdir` does *not* need to be specified on the command line if you are specifying this in the config file
3. `--user-agent` really should *not* be changed ... this *will* cause you issues in that format as it does not conform to the Microsoft OneDrive API requirements. If you *are* going to change it - update this in the `config` file only.



Can you provide the output of:
```
onedrive --confdir="/home/user/.Onedrive_Config/empresa/" --display-config
```
and 
```
onedrive --confdir="/home/user/.Onedrive_Config/personal/ --display-config
```

I think your issue stems from specifying a wrong path for `--confdir` in your commands.

**Reply by tcboeira** [reply]
Wow, how cool! What a show!
Yes, yes, I apologize, because I read and read again and could not find a solution.
In fact, it must be my misunderstanding. Maybe even understanding - or even confusion in the Online Translator - since I'm still learning your language.

But I did the same...
I created directories and placed the file inside each one, with the respective configuration
And the outputs were complete, with the settings I want!!!

Finally, when I'm synchronizing each account - personal and company - just repeat the following

`onedrive --synchronize --user-agent staff@outlook.com --confdir="/home/USER/.config/onedrive/personal"`
e
`onedrive --synchronize --user-agent user@company.com --confdir="/home/user/.config/onedrive/company"`

and the sync will be performed for the two accounts?

Thanks a lot for the help.

**Reply by abraunegg** [reply]
@tcboeira 
> `--user-agent staff@outlook.com`

Do not do this.

This will cause issues with Microsoft OneDrive API ...

> and the sync will be performed for the two accounts?

Correct.


1. Increase verbosity so you can *see* what the client is doing to understand what is occurring by adding `--verbose`
2. Test first by adding `--dry-run`

When comfortable with that the client will operate as per your expectation, remove `--dry-run` and sync.

Once all comfortable and running as per expected - drop the `--verbose`

**Reply by tcboeira** [reply]
Wave...
Well, before all else, I want to thank your help, the orientations that foram without definitive doubts in my questão.

As I mentioned, I created the configuration files for each scenario, a person and another company.

Apontei for each one and the results foram satisfactory...
- To see the settings
onedrive --confdir="/home/user/.config/onedrive/rsgovbr" --display-config
onedrive --confdir="/home/user/.config/onedrive/person" --display-config

Perform the first synchrony one of each time, or what happened with success
- To perform in sync
onedrive --synchronize --confdir="/home/user/.config/onedrive/rsgovbr"
onedrive --synchronize --confdir="/home/user/.config/onedrive/person"

Last year, I included the initialization of my system, two lines for each um, to keep monitoring and testing with absolute success.
- For permanent monitoring
onedrive --monitor --confdir="/home/user/.config/onedrive/rsgovbr"
onedrive --monitor --confdir="/home/user/.config/onedrive/person"


Bem, vejam, get success not what I want. I spent the last days testing synchrony, in different networks etc etc etc

I know for the same, and I really appreciate very much the help and attention that was given to me in this questão,

Big Hug!!!
